config/dune

```diff
@@ -9,5 +9,5 @@
 (rule
  (targets coq_config.ml)
  (mode fallback)
- (deps %{project_root}/configure.ml %{project_root}/dev/ocamldebug-coq.run)
- (action (chdir %{project_root} (run %{ocaml} configure.ml -local -warn-error yes -native-compiler yes))))
+ (deps %{project_root}/configure.ml %{project_root}/dev/ocamldebug-coq.run (env_var COQ_CONFIGURE_PREFIX))
+ (action (chdir %{project_root} (run %{ocaml} configure.ml -native-compiler no))))
```

Comment:
Why not use -prefix? Why drop native-compile?

Reply:
> Why not use -prefix?

That's a good question, the only way I've found to pass info to Dune is using configure variables. That being said, in Dune 1.3.0 we should be able to do `-prefix %{env:COQ_PREFIX}` and that will work, but current version cannot expand env vars as the environment needs to be threaded along the rules.

Anyways I am just being cautious here, maybe we could just set a fixed prefix here and that would work? The coqlib stuff is hard to understand for me.

> Why drop native-compile?

I can revert that, but I've found that it is misleading to compile Coq with native support enabled for now as `coq_dune` doesn't support it yet [support coming soon tho].

Reply:
Note that I am not excessively happy with this solution, which I see more of a workaround than a real solution until we can indeed mostly drop `-prefix` from coqlib location.

Reply:
Why does coq_dune need to do something wrt to native_compute?

Reply:
It needs to do the same than the makefile does. By default, even with native-compute on `coqc` won't generate native versions of the files, unless it is passed a special flag. Makefiles do setup this special flag, so they produce the `.coq_native` stuff, etc...

That should be easy to add, just need a bit of time.

Reply:
In fact I prefer to wait until `coq_dune` can read library config files so we can enable it on a per basis with `(mode native)`.